src/AppInstallerCLICore/Command.cpp

```diff
@@ -720,6 +722,25 @@ namespace AppInstaller::CLI
                 throw CommandException(Resource::String::TooManyArgError, arg.Name());
             }
         }
+
+        if (execArgs.Contains(Execution::Args::Type::Silent) && execArgs.Contains(Execution::Args::Type::Interactive))
+        {
+            throw CommandException(Resource::String::TooManyBehaviorsError, s_Command_ArgName_SilentAndInteractive);
+        }
+
+        if (execArgs.Contains(Execution::Args::Type::Manifest) &&
+            (execArgs.Contains(Execution::Args::Type::Query) ||
+             execArgs.Contains(Execution::Args::Type::Id) ||
+             execArgs.Contains(Execution::Args::Type::Name) ||
+             execArgs.Contains(Execution::Args::Type::Moniker) ||
+             execArgs.Contains(Execution::Args::Type::Version) ||
+             execArgs.Contains(Execution::Args::Type::Channel) ||
+             execArgs.Contains(Execution::Args::Type::Source) ||
+             execArgs.Contains(Execution::Args::Type::Exact) ||
+             execArgs.Contains(Execution::Args::Type::All)))
+        {
+            throw CommandException(Resource::String::BothManifestAndSearchQueryProvided, "");
+        }

```

Comment:
I don't really think this is the best place to put these validations, there may come a time in the future where we some command needs this to be false and then we have to move this to every command.  But I think it's ok to leave it for now; the better solution is parameter groups like PowerShell.

Reply:
Moved to individual commands

---
In reply to: [504132887](https://github.com/microsoft/winget-cli/pull/588#discussion_r504132887) [](ancestors = 504132887)